flow/designs/gf180/uart-blocks/tapcell.tcl

```diff
@@ -1,8 +1,13 @@
-  tapcell \
-    -endcap_cpp             "12" \
-    -distance               100 \
-    -tapcell_master         $::env(TIE_CELL) \
-    -endcap_master          $::env(ENDCAP_CELL) \
-    -halo_width_x $::env(MACRO_HALO_X) \
-    -halo_width_y $::env(MACRO_HALO_Y)
+if {[info exists ::env(TAP_CELL_NAME)] && $::env(TAP_CELL_NAME) != "" &&
+    [info exists ::env(ENDCAP_CELL_NAME)] && $::env(ENDCAP_CELL_NAME) != ""} {
+    tapcell \
+	-endcap_cpp             "12" \
+	-distance               100 \
+	-tapcell_master         $::env(TAP_CELL_NAME) \
+	-endcap_master          $::env(ENDCAP_CELL_NAME) \
+	-halo_width_x $::env(MACRO_HALO_X) \
+	-halo_width_y $::env(MACRO_HALO_Y)
+} else {
+    puts "Skipping tapcell insertion since TAP_CELL_NAME and/or ENDCAP_CELL_NAME are not defined"
```

Comment:
Would recommend flagging this as a warning because this is not expected behavior

Reply:
Yeah, I wasn't sure if there was a "warn" function that's exposed to ORFS, but it looks like there is. It looks like I need to:

1. Define the new TAP warning in src/tap/messages.txt
2. call utl::warn with the TAP + code in each of the tapcell.tcl's with the warning

Does that sound about right?

Reply:
@maliberty I'm inclined to disagree.

There's lots of stuff that would have to be warned off in that case, for instance setting "GND_NETS_VOLTAGES" to an empty string.

This warning will then be a wolf-wolf for real warnings to care about even with this simplification(to increase turnaround times during development).

I can't imagine that this option will be forgotten before a tapeout.

Does it have any observable effects, except in the event of a tapeout?

Reply:
Fill cells won't; tapcells do somewhat affect placement.  utl::warn/utl::info are available in TCL.  I'm ok with info or warn.